uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/PortletsRESTControllerTest.java

```diff
@@ -209,41 +217,31 @@ private void givenPortletDefinitionsInRegistry(IPortletDefinition... portletDefi
 
     private void givenUserHasPermissionForPortlets(
             IPerson user,
-            PortletsRESTController.PortletPermissionType permissionType,
+            IAuthorizationService.PortletPermissionType permissionType,
             IPortletDefinition... portletDefinitions) {
         for (IPortletDefinition portletDefinition : portletDefinitions) {
             final String portletDefinitionStringId =
                     portletDefinition.getPortletDefinitionId().getStringId();
             switch (permissionType) {
                 case BROWSE:
-                    given(
-                                    this.authorizationService.canPrincipalBrowse(
-                                            this.authorizationPrincipal, portletDefinition))
-                            .willReturn(true);
+                    this.authorizationService.canPrincipalBrowse(
+                            this.authorizationPrincipal, portletDefinition);
                     break;
                 case CONFIGURE:
-                    given(
-                                    this.authorizationService.canPrincipalConfigure(
-                                            this.authorizationPrincipal, portletDefinitionStringId))
-                            .willReturn(true);
+                    this.authorizationService.canPrincipalConfigure(
+                            this.authorizationPrincipal, portletDefinitionStringId);
                     break;
                 case MANAGE:
-                    given(
-                                    this.authorizationService.canPrincipalManage(
-                                            this.authorizationPrincipal, portletDefinitionStringId))
-                            .willReturn(true);
+                    this.authorizationService.canPrincipalManage(
+                            this.authorizationPrincipal, portletDefinitionStringId);
                     break;
                 case SUBSCRIBE:
-                    given(
-                                    this.authorizationService.canPrincipalSubscribe(
-                                            this.authorizationPrincipal, portletDefinitionStringId))
-                            .willReturn(true);
+                    this.authorizationService.canPrincipalSubscribe(
+                            this.authorizationPrincipal, portletDefinitionStringId);
                     break;
                 case RENDER:
-                    given(
-                                    this.authorizationService.canPrincipalRender(
-                                            this.authorizationPrincipal, portletDefinitionStringId))
-                            .willReturn(true);
+                    this.authorizationService.canPrincipalRender(
+                            this.authorizationPrincipal, portletDefinitionStringId);
```

Comment:
Shouldn't these tests assert something?
Why has the testing part of the test been removed?

Reply:
This was previously handled in a different way as this same switch case with method calls was used in the class. Now since that is now handled  and replaced with a single polymorphic call, I did not need this any more in my test. Hence, I modified the existing flow to test whether the my polymorphism methods are being invoked properly.